python/lsst/obs/metadata/translator.py

```diff
@@ -0,0 +1,252 @@
+# This file is part of obs_metadata.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Classes and support code for metadata translation"""
+
+__all__ = ("MetadataTranslator",)
+
+from abc import abstractmethod
+import logging
+import math
+
+import astropy.units as u
+
+log = logging.getLogger(__name__)
+
+
+class MetadataMeta(type):
+    """Register all subclasses with the base class"""
+
+    @staticmethod
+    def _makeConstMapping(standardKey, constant):
+        def constant_translator(self):
+            return constant
+        constant_translator.__doc__ = f"""Returns constant value for '{standardKey}' property
+
+        Returns
+        -------
+        translation : `{type(constant).__name__}`
+            Always returns {constant!r}.
+        """
+        return constant_translator
+
+    @staticmethod
+    def _makeTrivialMapping(standardKey, fitsKey, default=None, minimum=None, maximum=None, unit=None):
+        def trivial_translator(self):
+            if unit is not None:
+                return self.quantity_from_card(fitsKey, unit,
```

Comment:
This assumes `quantity_from_card` is defined on `self`: is there a way to enforce this better (and any other metaclass/ABC assumptions)?

Reply:
I think `quantity_from_card` is in the base class already so should always be available.